static/js/compose.js

```diff
@@ -475,13 +475,25 @@ function validate_stream_message_announce(stream_name) {
     return true;
 }
 
-function validate_stream_message_announcement_only(stream_name) {
-    // Only allow realm admins to post to announcement_only streams.
-    const is_announcement_only = stream_data.get_announcement_only(stream_name);
-    if (is_announcement_only && !page_params.is_admin) {
+function validate_stream_message_post_policy(stream_name) {
+    const stream_post_policy = stream_data.get_stream_post_policy(stream_name);
+    if (stream_post_policy === 2 && !page_params.is_admin) {
         compose_error(i18n.t("Only organization admins are allowed to post to this stream."));
         return false;
     }
+    const person = people.get_person_from_user_id(page_params.user_id);
+    const current_datetime = new Date(Date.now());
+    const person_date_joined = new Date(person.date_joined);
+    const days = new Date(current_datetime - person_date_joined).getDate();
+
+    if (!page_params.is_admin) {
+        if (stream_post_policy === 3 && days < page_params.realm_waiting_period_threshold) {
+            let error_text = "New members are not allowed to post to this stream. ";
+            error_text += "You can send messages only after " + days + " days.";
```

Comment:
Constructing error text this way doesn't work with i18n.  Read https://zulip.readthedocs.io/en/latest/translating/internationalization.html to learn about how this works, but roughly, you'll want to use `i8n.t("New members ... __days__ ...", days)`.